bindings/python/tests/test_parse.py

```diff
@@ -0,0 +1,237 @@
+from context import scripts
+import scripts.parse as parse
+
+import json
+import os
+
+import clang.cindex as clang
+
+
+def create_compilation_database(tmp_path, filepath):
+    input = tmp_path / "compile_commands.json"
+    x = [
+        {
+            "directory": f"{tmp_path}",
+            "command": f"/usr/bin/clang++ -std=c++14 {filepath}",
+            "file": f"{filepath}",
+        }
+    ]
+    input.write_bytes(str(x).encode())
+    tmp_path = str(tmp_path)
+    return tmp_path
+
+
+def get_parsed_info(tmp_path, file_contents):
+    source_path = tmp_path / "file.hpp"
+    source_path.write_bytes(str(file_contents).encode())
+
+    parsed_info = parse.parse_file(
+        source=str(source_path),
+        compilation_database_path=create_compilation_database(
+            tmp_path=tmp_path, filepath=source_path
+        ),
+    )
+
+    return parsed_info
+
+
+def test_types(tmp_path):
+    file_contents = ""
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    assert type(parsed_info) is dict
+    assert type(parsed_info["members"]) is list
+
+
+def test_translation_unit(tmp_path):
+    file_contents = ""
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    assert parsed_info["kind"] == "TRANSLATION_UNIT"
+
+
+def test_namespace(tmp_path):
+    file_contents = "namespace Anamespace {}"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    namespace = parsed_info["members"][0]
+
+    assert namespace["kind"] == "NAMESPACE"
+
+
+def test_namespace_ref(tmp_path):
+    file_contents = "#include<ostream> \n std::ostream Aostream;"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    namespace_ref = parsed_info["members"][0]["members"][0]
+
+    assert namespace_ref["kind"] == "NAMESPACE_REF"
+
+
+def test_struct_decl(tmp_path):
+    file_contents = "struct AStruct{};"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+
+    assert struct_decl["kind"] == "STRUCT_DECL"
+
+
+def test_cxx_base_specifier(tmp_path):
+    file_contents = "struct AStruct{}; struct BStruct: public AStruct{};"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    child_struct_decl = parsed_info["members"][1]
```

Comment:
A rephrasing would be better
```py
for member in parsed_info['members']:
  if member['name'] == 'Derived':
    child_struct_decl = member
    break
```

Try to show the assumption without needing to make leaps of faith

Reply:
That was my initial approach, but @aponza advised to just use index since we know exactly how it has to be. I am in favour of the `for loop` approach, even though libclang is pretty stable.

Some input from @aponza?

Reply:
I try to follow the best practice of not testing the implementation, be it yours or theirs. In this case a `contains`+`get` would do wonders, you can implement it with Kunal's for loop or with something like [`any`](https://thispointer.com/python-how-to-check-if-an-item-exists-in-list-search-by-value-or-condition/): `child_struct_decl = any(member['name'] == 'Derived' for member in parsed_info['members']`) should be the line, right?

Also note I'd change `AStruct` - `BStruct`to `ABaseStruct`/`AParentStruct` - `ADerivedStruct`/`AChildStruct` and in general, the prefixing a/an is to be more descriptive of what thing I'm using. I don't know if it's a widespread practice, it's what I do myself, so variables could be `a_very_long_file_hash` or ,_expected_json`, ...

Reply:
I'd definitely prefer `ADerivedStruct` over `Derived` (my example). My reasoning for using a loop:
* I don't know if the implementing library guarantees ordered AST

If `clang` guarantees that, then no need and my suggestion is verbose because I'm new to the library

Reply:
> If clang guarantees that

How do we check that?

Here's the function in clang:
```py
    def get_children(self):
        """Return an iterator for accessing the children of this cursor."""

        def visitor(child, parent, children):
            assert child != conf.lib.clang_getNullCursor()

            # Create reference to TU so it isn't GC'd before Cursor.
            child._tu = self._tu
            children.append(child)
            return 1 # continue
        children = []
        conf.lib.clang_visitChildren(self, callbacks['cursor_visit'](visitor),
            children)
        return iter(children)
```

Reply:
Dunno. Maybe note the assumption in case the test fails in future?